lib/fluent/plugin/out_tdlog.rb

```diff
@@ -67,19 +62,12 @@ def configure(conf)
         end
       end
 
-      if conf.has_key?('tmpdir')
-        @tmpdir = conf['tmpdir']
-        FileUtils.mkdir_p(@tmpdir)
-      end
+      FileUtils.mkdir_p(@tmpdir) if @tmpdir
 
-      database = conf['database']
-      table = conf['table']
-      if database && table
-        validate_database_and_table_name(database, table, conf)
-        @key = "#{database}.#{table}"
+      if @database && @table
```

Comment:
Ah, I made a fault in last comment... "Either @database or @table are NOT configured".
Anyway, it's nice to add tests too.